dali/python/nvidia/dali/_debug_mode.py

```diff
@@ -0,0 +1,332 @@
+        return _PipelineDebug.current()._wrap_op_call(_arithm_op, "sub", other, self)
+
+    def __mul__(self, other):
+        return _PipelineDebug.current()._wrap_op_call(_arithm_op, "mul", self, other)
+
+    def __rmul__(self, other):
+        return _PipelineDebug.current()._wrap_op_call(_arithm_op, "mul", other, self)
+
+    def __pow__(self, other):
+        return _PipelineDebug.current()._wrap_op_call(_arithm_op, "pow", self, other)
+
+    def __rpow__(self, other):
+        return _PipelineDebug.current()._wrap_op_call(_arithm_op, "pow", other, self)
+
+    def __truediv__(self, other):
+        return _PipelineDebug.current()._wrap_op_call(_arithm_op, "fdiv", self, other)
+
+    def __rtruediv__(self, other):
+        return _PipelineDebug.current()._wrap_op_call(_arithm_op, "fdiv", other, self)
+
+    def __floordiv__(self, other):
+        return _PipelineDebug.current()._wrap_op_call(_arithm_op, "div", self, other)
+
+    def __rfloordiv__(self, other):
+        return _PipelineDebug.current()._wrap_op_call(_arithm_op, "div", other, self)
+
+    def __neg__(self):
+        return _PipelineDebug.current()._wrap_op_call(_arithm_op, "minus", self)
+
+    def __eq__(self, other):
+        return _PipelineDebug.current()._wrap_op_call(_arithm_op, "eq", self, other)
+
+    def __ne__(self, other):
+        return _PipelineDebug.current()._wrap_op_call(_arithm_op, "neq", self, other)
+
+    def __lt__(self, other):
+        return _PipelineDebug.current()._wrap_op_call(_arithm_op, "lt", self, other)
+
+    def __le__(self, other):
+        return _PipelineDebug.current()._wrap_op_call(_arithm_op, "leq", self, other)
+
+    def __gt__(self, other):
+        return _PipelineDebug.current()._wrap_op_call(_arithm_op, "gt", self, other)
+
+    def __ge__(self, other):
+        return _PipelineDebug.current()._wrap_op_call(_arithm_op, "geq", self, other)
+
+    def __and__(self, other):
+        return _PipelineDebug.current()._wrap_op_call(_arithm_op, "bitand", self, other)
+
+    def __rand__(self, other):
+        return _PipelineDebug.current()._wrap_op_call(_arithm_op, "bitand", other, self)
+
+    def __or__(self, other):
+        return _PipelineDebug.current()._wrap_op_call(_arithm_op, "bitor", self, other)
+
+    def __ror__(self, other):
+        return _PipelineDebug.current()._wrap_op_call(_arithm_op, "bitor", other, self)
+
+    def __xor__(self, other):
+        return _PipelineDebug.current()._wrap_op_call(_arithm_op, "bitxor", self, other)
+
+    def __rxor__(self, other):
+        return _PipelineDebug.current()._wrap_op_call(_arithm_op, "bitxor", other, self)
+
+
+class _PipelineDebugRunner():
+    def __init__(self, pipeline):
+        self.pipeline = pipeline
+
+    def __enter__(self):
+        self.pipeline._debug_on = True
+        self.pipeline._external_source_debug = True
+        self.pipeline._cur_subpipeline_id = -1
+        return super(type(self.pipeline), self.pipeline).__enter__()
```

Comment:
👀

Reply:
I just like using `with` but I guess it's not necessary.

Reply:
I'm fine either way, was a bit surprised by the `super(type(self.pipeline)`.